src/features/auth/SignUp.jsx

```diff
@@ -0,0 +1,170 @@
+import { userSignUp } from "./authSlice";
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { useForms } from "./useForms";
+import { useDispatch, useSelector } from "react-redux";
+
+export const SignUp = () => {
+  const [userDetails, setUserDetails] = useState({
+    firstName: "",
+    lastName: "",
+    userName: "",
+    email: "",
+    password: "",
+  });
+
+  const token = useSelector((state) => state.auth.token);
+  const dispatch = useDispatch();
+  const signUpStatus = useSelector((state) => state.auth.signUpStatus);
+  const error = useSelector((state) => state.auth.error);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (signUpStatus === "succeeded") {
+      localStorage?.setItem("login", JSON.stringify({ token }));
+      navigate("/");
+    }
+  }, [signUpStatus, navigate, token]);
+
+  const signUp = async () => {
+    if (signUpStatus === "idle" || signUpStatus === "failed") {
+      dispatch(userSignUp(userDetails));
+    }
+  };
+
+  const {
+    touchedFields,
+    handleOnChange,
+    handleOnBlur,
+    validateSignUp,
+    isBtnDisabled,
+  } = useForms(setUserDetails);
+
+  function handleSignUp(e) {
+    e.preventDefault();
+
+    signUp();
+  }
+  const errors = validateSignUp(userDetails);
+
+  const shouldShowErrors = (field) => {
+    return errors[field] ? touchedFields[field] : false;
+  };
+
+  const getInputClassName = (field) => {
+    return shouldShowErrors(field)
+      ? "text-input error-border-color"
+      : "text-input generic-border-color";
+  };
+  return (
+    <div className="center-page-align auth-form-size">
+      <form
+        onSubmit={handleSignUp}
+        className=" border-all gray-border padding-all"
+        noValidate
+      >
+        <div className="font-size-3 margin-bottom text-center">Sign Up</div>
+        <div class="flex-column margin-bottom">
```

Comment:
I think you have used class instead of className in many places, you can change that.